### PR TITLE
[NEW] Add support for attachment fields + some visual improvements

### LINKED ISSUE #1231: [NEW] Only first attachment field gets displayed
- Your Rocket.Chat app version: 1.8.0
- Your Rocket.Chat server version: 0.61.0
- Device (or Simulator) you're running with: iPhone 7+

We use Rocket Chat in Connection with our Icinga Monitoring Platform. I recently updated our script to use the message attachments. This works great on the Desktop App. Unfortunately the iOS App only shows the first attachment field.

Btw.: The Emoji Tag doesn't seem to work on mobile as well.

Here is some sample Json we send to the hook:
*The `${...}` are variables in the shell script...*

```
{  
   "emoji":"${ICON}",
   "text":"*${NOTIFICATIONTYPE}* ${SERVICEDISPLAYNAME} ( _${HOSTDISPLAYNAME}_ ): *${SERVICESTATE}* | ${SERVICEOUTPUT} ",
   "attachments":[  
      {  
         "color":"${COLOR}",
         "fields":[  
            {  
               "short":"true",
               "title":"State",
               "value":"${SERVICESTATE}"
            },
            {  
               "short":"true",
               "title":"Host",
               "value":"${HOSTDISPLAYNAME}"
            },
            {  
               "short":"false",
               "title":"Service",
               "value":"${SERVICEDISPLAYNAME}"
            },
            {  
               "short":"false",
               "title":"Check Output",
               "value":"${SERVICEOUTPUT}"
            }
         ]
      }
   ]
}
```

**It looks fine in the Desktop Application**
icinga_desktop

*The iOS Application shows only the first field even when you tap on it.*
icinga_mobile

I'm happy to help if you guys need more information.

Thanks in advance, 
Maurice

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

|Preview|
|-|
||

Closes #1231 

I sent some test messages with fields at: https://open.rocket.chat/channel/cardoso_test

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift [hunk 1]
```diff
@@ -112,7 +112,7 @@ extension ChatViewController: ChatMessageCellProtocol {
         openDocument(attachment: attachment)
     }
 
-    func viewDidCollpaseChange(view: UIView) {
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChatControllerMessageCellProtocol.swift]: Good catch! :-)

File: Rocket.Chat/Models/Mapping/AttachmentModelMapping.swift [hunk 2]
```diff
@@ -10,6 +10,18 @@ import Foundation
...
+    func map(_ values: JSON, realm: Realm?) {
+        if self.identifier == nil {
+            self.identifier = String.random()
+        }
+
+        self.short = values["short"].boolValue
+        self.title = values["title"].stringValue
+        self.value = values["value"].stringValue
+    }
+}
```
Comment (rafaelks) [inline Rocket.Chat/Models/Mapping/AttachmentModelMapping.swift]: 👍

**Comment by cardoso** [comment]
@filipealva this was a tricky one. Take a look again when you can. Thanks!

**Comment by filipealva** [comment]
@cardoso yes, it's a tricky one :( 

I just took a look again and it seems that the issue is still happening. 

1.  Happening without expanding/collapsing (just scrolling): 


2. Happing while expanding/collapsing + scrolling:

**Comment by rafaelks** [comment]
@cardoso Maybe this is something on the reuse of the cell.

**Pull request merged**